### PR TITLE
Tracing: Allow trace to logs for OpenSearch

### LINKED ISSUE #79: Opensearch: Add trace to logs functionality
### Summary

Add trace to logs functionality for OS to allow OS users jump into their logs from tempo traces. 

### Problem statement

Currently, OS is not supported in trace to logs functionality. We want to add logic so OS is supported for this feature. 
 
### Scope

Implementation

### Solution

This should be implemented after ES implementation.

This is hardcoded for splunk + loki here https://github.com/grafana/grafana/blob/2970c6db375e258a6fbf129c21b7210af248d40c/public/app/features/explore/TraceView/createSpanLink.tsx#L115
It just needs a case for elastic so it creates correct query.

And then in `public/app/core/components/TraceToLogs/TraceToLogsSettings.tsx` elasticsearch should be added.

### Resources
Discussed in https://github.com/grafana/grafana/discussions/51746

### PR DESCRIPTION
**Pull request by gabor**
You can configure a tracing datasource to allow users to jump from traces to logs stored in [OpenSearch](https://github.com/grafana/opensearch-datasource)

(this pull requests builds on the changes that happened in https://github.com/grafana/grafana/pull/58063 )

fixes https://github.com/grafana/opensearch-datasource/issues/79

an example screenshot, achieved by running a tempo datasource, with label-mappings (`geo_dst => geo.dest`, `geo_src => geo.src`):
tracing

### CODE REVIEW

File: public/app/features/explore/TraceView/createSpanLink.tsx [hunk 1]
```diff
@@ -283,7 +286,17 @@ function getLinkForLoki(span: TraceSpan, options: TraceToLogsOptions, dataSource
...
+// so here is a minimal interface that handles both elasticsearch and opensearch.
+interface ElasticSearchOrOpensearchQuery extends DataQuery {
+  query: string;
+  metrics: Array<{
+    id: string;
+    type: 'logs';
+  }>;
+}
+
+function getLinkForElasticsearchOrOpensearch(
```
Comment (joey-grafana) [inline public/app/features/explore/TraceView/createSpanLink.tsx:299]: nit: we should use either `ElasticSearch` or `Elasticsearch` :)

File: public/app/features/explore/TraceView/createSpanLink.tsx [hunk 2]
```diff
@@ -116,7 +115,11 @@ function legacyCreateSpanLinkFactory(
           dataLink = getLinkForSplunk(span, traceToLogsOptions, logsDataSourceSettings);
           break;
         case 'elasticsearch':
-          dataLink = getLinkForElasticsearch(span, traceToLogsOptions, logsDataSourceSettings);
+          dataLink = getLinkForElasticsearchOrOpensearch(span, traceToLogsOptions, logsDataSourceSettings);
+          break;
+        case 'grafana-opensearch-datasource':
+          dataLink = getLinkForElasticsearchOrOpensearch(span, traceToLogsOptions, logsDataSourceSettings);
```
Comment (joey-grafana) [inline public/app/features/explore/TraceView/createSpanLink.tsx:121]: nit: we're setting dataLink to the same value in both cases. Can we merge them?

**Review by joey-grafana — APPROVED** [review]

**Review by ivanahuckova — APPROVED** [review]
+1 to what @joey-grafana said.

File: public/app/features/explore/TraceView/createSpanLink.tsx (hunk 1 above)
Comment (gabor) [reply public/app/features/explore/TraceView/createSpanLink.tsx]: good catch! i checked and "Elasticsearch" seems to be the right way to capitalize it, so i did it that way 👍

File: public/app/features/explore/TraceView/createSpanLink.tsx (hunk 2 above)
Comment (gabor) [reply public/app/features/explore/TraceView/createSpanLink.tsx:121]: you are right, but i personally don't like fall-through case-statements.. but it's your squad's codebase, so if you prefer to do it so, i can adjust it. just to clarify, is this what you'd prefer to have?:
```typescript
switch(...) {
    case 'loki':
        datalink = ...
        break;
    case 'splunk':
       datalink = ...
       break;
    case 'elasticsearch':
    case 'opensearch':
        datalink = ....
        break;
}
```
?

Comment (joey-grafana) [reply public/app/features/explore/TraceView/createSpanLink.tsx:121]: Ahh this is no problem. That is what I head in mind. I'm happy with either as it's no big deal :)

**Pull request merged**